src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -170,6 +171,7 @@ public int GetHashCode(object obj)
     }
 
     [Serializable]
+    [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
     internal sealed class CultureAwareComparer : StringComparer
```

Comment:
I'm mostly concerned with the long term maintenance and the general notion of exposing internals. I think the general binary serialization support is already going limit the set of changes we can make so I will just have to get over that. 

My concerns shouldn't block moving forward on this I just wanted to make sure folks understand what we are doing. @ViktorHofer I can help you figure out what needs to happen to accomplish this. Perhaps we can come up with a solution that doesn't require us to maintain an entire mscorlib ref assembly in corefx to expose these. We can probably make some changes to GenFacades to accomplish this.